graph/pull.go

```diff
@@ -665,21 +676,27 @@ func (s *TagStore) pullV2Tag(r *registry.Session, out io.Writer, endpoint *regis
 						Action:    "Extracting",
 					}))
 				if err != nil {
-					return false, err
+					return err
 				}
-
 				if err := s.graph.SetDigest(d.img.ID, d.digest); err != nil {
-					return false, err
+					return err
 				}
-
+				out.Write(sf.FormatProgress(stringid.TruncateID(d.img.ID), "Pull complete", nil))
+				tagUpdated = true
 				// FIXME: Pool release here for parallel tag pull (ensures any downloads block until fully extracted)
+			} else {
+				out.Write(sf.FormatProgress(stringid.TruncateID(d.img.ID), "Already exists", nil))
 			}
-			out.Write(sf.FormatProgress(stringid.TruncateID(d.img.ID), "Pull complete", nil))
-			tagUpdated = true
-		} else {
-			out.Write(sf.FormatProgress(stringid.TruncateID(d.img.ID), "Already exists", nil))
+			return nil
 		}
+		err := registerFunc(&downloads[i], lastErr != nil)
+		if err != nil {
+			lastErr = err
```

Comment:
This is potentially returning a different error than before. Before the first error returned was the returned error, now it will be the last error.